### PR TITLE
Add error checking

### PR DESCRIPTION
**Pull request by almasaeed2010**
- Adds an example XML file with an error element in `examples/error.xml`
- Adds 2 new methods to `EUtilsResource` class
   - `hasError(): bool`. Whether the response has an error element
   - `errorMessage(): string`. The error message from the response 
- `isSuccessful()` returns false if an error element exists

### CODE REVIEW

File: includes/resources/EUtilsResource.inc [hunk 1]
```diff
@@ -53,9 +60,41 @@ class EUtilsResource {
...
+      return TRUE;
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * Get the error message.
+   *
+   * @return string|null The message or null if none exist.
```
Comment (almasaeed2010) [inline includes/resources/EUtilsResource.inc]:
```suggestion
   * @return string|null 
   *   The message or null if none exist.
```

File: includes/resources/EUtilsResource.inc [hunk 2]
```diff
@@ -53,9 +60,42 @@ class EUtilsResource {
...
+      }
+    }
+
     return $status >= 200 && $status <= 299;
   }
 
+  /**
+   * Find and set errors.
+   *
+   * @return bool
```
Comment (almasaeed2010) [inline includes/resources/EUtilsResource.inc]:
```suggestion
   * @return bool
   *   Whether the response has an error element.
```

File: includes/resources/EUtilsResource.inc [hunk 3]
```diff
@@ -53,9 +60,42 @@ class EUtilsResource {
...
+      return TRUE;
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * Get the error message.
+   *
+   * @return string|null
```
Comment (almasaeed2010) [inline includes/resources/EUtilsResource.inc]:
```suggestion
   * @return string|null
```

File: includes/resources/EUtilsResource.inc [hunk 4]
```diff
@@ -73,7 +132,13 @@ class EUtilsResource {
    *   The response in XML.
    */
   public function xml() {
-    return simplexml_load_string($this->originalBody());
+    // Check cache
```
Comment (almasaeed2010) [inline includes/resources/EUtilsResource.inc]:
```suggestion
    // Check cache.
```

File: includes/resources/EUtilsResource.inc [hunk 5]
```diff
@@ -73,7 +132,13 @@ class EUtilsResource {
...
+    if (!is_null($this->xml)) {
+      return $this->xml;
+    }
+
+    // Cache xml
```
Comment (almasaeed2010) [inline includes/resources/EUtilsResource.inc]:
```suggestion
    // Cache xml.
```

**Comment by almasaeed2010** [comment]
So now I added new checks to account for most cases. I did a functional test of all DBs and we catch the error correctly.

**Comment by bradfordcondon** [comment]
awesome, thank you very much.

**Comment by bradfordcondon** [comment]
heres the accession for the failing request

```
array(2) {
  [0] =>
  string(10) "bioproject"
  [1] =>
  string(6) "291087"
}

```

In all likelihood hte test should be making requests and trying to insert linked records, for starters...

**Comment by almasaeed2010** [comment]
There was a stupid mistake where I use `== 0` when I should've used `=== 0`. Problem fixed.

**Pull request merged**